### PR TITLE
Parametrize atlas security users

### PR DESCRIPTION
**Pull request by anatbal**
- Removed hardcoded admin, ohdsi users
- Generate password for admin user and save in kv
- Allow the user to add a comma delimited list of users

### CODE REVIEW

File: infra/main.bicep [hunk 1]
```diff
@@ -160,23 +176,19 @@ resource deploymentAtlasSecurity 'Microsoft.Resources/deploymentScripts@2020-10-
         secureValue: 'host=${atlasDatabase.outputs.postgresServerFullyQualifiedDomainName} port=5432 dbname=${atlasDatabase.outputs.postgresWebApiDatabaseName} user=${atlasDatabase.outputs.postgresWebapiAdminUsername} password=${postgresWebapiAdminPassword} sslmode=require'
       }
       {
-        name: 'SQL'
-        value: loadTextContent('sql/atlas_security.sql')
+        name: 'ATLAS_SECURITY_ADMIN_PASSWORD'
+        secureValue: atlasSecurityAdminPassword
+      }
+      {
+        name: 'ATLASUSERS'
```
Comment (tamirkamara) [inline infra/main.bicep:183]:
```suggestion
        name: 'ATLAS_USERS'
```

File: infra/scripts/atlas_security.sh [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    username=$i
+    let count+=1
+    continue
+else
+    userid=`psql -t -v ON_ERROR_STOP=1 "$OHDSI_ADMIN_CONNECTION_STRING" -c "select id from webapi.sec_user where login = '$username';"`
+    if [ "$count" = "2" ]; then
+    psql -v ON_ERROR_STOP=1 -e "$OHDSI_ADMIN_CONNECTION_STRING" -c "insert into webapi.sec_user_role (user_id, role_id) values ($userid,2);"
+    psql -v ON_ERROR_STOP=1 -e "$OHDSI_ADMIN_CONNECTION_STRING" -c "insert into webapi.sec_user_role (user_id, role_id) values ($userid,10);"
+    firstuser=$username
+    firstpassword="$i"
```
Comment (tamirkamara) [inline infra/scripts/atlas_security.sh:42]: where is this used?

File: infra/scripts/atlas_security.sh [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    atlaspw=`htpasswd -bnBC 4 "" $i | tr -d ':\n' | sed 's/$2y/$2a/'`
+    psql -v ON_ERROR_STOP=1 -e "$OHDSI_ADMIN_CONNECTION_STRING" -c "insert into webapi_security.security (email,password) values ('$username', E'$atlaspw');"
+    psql -v ON_ERROR_STOP=1 -e "$OHDSI_ADMIN_CONNECTION_STRING" -c "insert into webapi.sec_user (id, login, name) values ($id, '$username', '$username') ON CONFLICT DO NOTHING;"
+    let count+=1
+    let id+=1
+fi
+done
+
+count=1
+for i in $(echo $ATLASUSERS | sed "s/,/ /g")
```
Comment (tamirkamara) [inline infra/scripts/atlas_security.sh:30]: Wouldn't is be easier to do this work in a single loop (above)?

File: infra/scripts/atlas_security.sh [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+do
+if [ `expr $count % 2` -eq "1" ]; then
+    username=$i
+    let count+=1
+    continue
+else
+    userid=`psql -t -v ON_ERROR_STOP=1 "$OHDSI_ADMIN_CONNECTION_STRING" -c "select id from webapi.sec_user where login = '$username';"`
+    if [ "$count" = "2" ]; then
+    psql -v ON_ERROR_STOP=1 -e "$OHDSI_ADMIN_CONNECTION_STRING" -c "insert into webapi.sec_user_role (user_id, role_id) values ($userid,2);"
+    psql -v ON_ERROR_STOP=1 -e "$OHDSI_ADMIN_CONNECTION_STRING" -c "insert into webapi.sec_user_role (user_id, role_id) values ($userid,10);"
```
Comment (tamirkamara) [inline infra/scripts/atlas_security.sh:40]: Something isn't right about the roles you assign. This doesn't look comparable with my original script.

**Review by tamirkamara — CHANGES_REQUESTED** [review]

File: infra/main.bicep (hunk 1 above)
Comment (anatbal) [reply infra/main.bicep]: Done

File: infra/scripts/atlas_security.sh (hunk 2 above)
Comment (anatbal) [reply infra/scripts/atlas_security.sh]: Removed

File: infra/scripts/atlas_security.sh (hunk 3 above)
Comment (anatbal) [reply infra/scripts/atlas_security.sh]: Fixed

File: infra/scripts/atlas_security.sh (hunk 4 above)
Comment (anatbal) [reply infra/scripts/atlas_security.sh]: Fixed

File: infra/scripts/atlas_security.sh [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    username=$i
+    let count+=1
+    continue
+else
+    atlaspw=`htpasswd -bnBC 4 "" $i | tr -d ':\n' | sed 's/$2y/$2a/'`
+    psql -v ON_ERROR_STOP=1 -e "$OHDSI_ADMIN_CONNECTION_STRING" -c "insert into webapi_security.security (email,password) values ('$username', E'$atlaspw');"
+    psql -v ON_ERROR_STOP=1 -e "$OHDSI_ADMIN_CONNECTION_STRING" -c "insert into webapi.sec_user (id, login, name) values ($id, '$username', '$username') ON CONFLICT DO NOTHING;"
+    psql -v ON_ERROR_STOP=1 -e "$OHDSI_ADMIN_CONNECTION_STRING" -c "insert into webapi.sec_user_role (user_id, role_id) values ($id,1);" #public role
+    if [ "$count" = "2" ]; then
+    psql -v ON_ERROR_STOP=1 -e "$OHDSI_ADMIN_CONNECTION_STRING" -c "insert into webapi.sec_user_role (user_id, role_id) values ($id,2);" #admin role
```
Comment (tamirkamara) [inline infra/scripts/atlas_security.sh:26]:
```suggestion
        psql -v ON_ERROR_STOP=1 -e "$OHDSI_ADMIN_CONNECTION_STRING" -c "insert into webapi.sec_user_role (user_id, role_id) values ($id,2);" #admin role
```

File: infra/scripts/atlas_security.sh [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+    else
+    psql -v ON_ERROR_STOP=1 -e "$OHDSI_ADMIN_CONNECTION_STRING" -c "insert into webapi.sec_user_role (user_id, role_id) values ($id,10);"  #atlas user role
```
Comment (tamirkamara) [inline infra/scripts/atlas_security.sh:28]:
```suggestion
        psql -v ON_ERROR_STOP=1 -e "$OHDSI_ADMIN_CONNECTION_STRING" -c "insert into webapi.sec_user_role (user_id, role_id) values ($id,10);"  #atlas user role
```

**Review by tamirkamara — APPROVED** [review]

**Pull request merged**